portal-impl/src/com/liferay/portal/service/impl/ResourcePermissionLocalServiceImpl.java

```diff
@@ -362,8 +362,11 @@ public ResourcePermission fetchResourcePermission(
 			resourcePermissions.size());
 
 		for (ResourcePermission resourcePermission : resourcePermissions) {
+			Set<String> emptySet = Collections.emptySet();
+
 			if (resourcePermission.getActionIds() == 0) {
-				continue;
+				roleIdsToActionIds.put(
```

Comment:
1) Is it really intentional to not continue here? 

2) Also you have a wasteful assignment. The var is never used a second time so you could just do:

```java
roleIdsToActionIds.put(
    resourcePermission.getRoleId(), Collections.emptySet());
```